docs/2-advanced/09-accessibility.md

```diff
@@ -63,6 +63,72 @@ Theming is an important aspect when it comes to a UI5 Web Components application
 
 For more information regarding the available themes and how to use them, see the [Configuration](../2-advanced/01-configuration.md) section.
 
+### Theme Selection According to OS Settings
+
+
+The UI5 Web Components framework does not offer a built-in mechanism for selecting themes based on the users' OS settings. However, we recommend using standard APIs to implement OS-based theme selection in applications built with UI5 Web Components.
+
+#### Light | Dark
+
+To synchronize theme switching with the OS's light or dark mode, you can use the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) CSS Media feature, as shown in the next example:
+
+Check `prefers-color-scheme` for `dark` or `light` and apply one of the availabe light/dark themes (Horizon Morning, Horizon Evening, ect.)
+
+```ts
+    import { setTheme } from "@ui5/webcomponents-base/dist/config/Theme.js";
+
+	const darkMode = window.matchMedia("(prefers-color-scheme: dark)").matches;
+
+	setTheme(darkMode ? "sap_horizon_dark" : "sap_horizon");
+```
+
+#### Contrast 
+
+To switch to a high contrast theme when the OS does, you can use [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme), [forced-colors](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors) CSS Media features (detecting Windows High Contrast mode) and  [prefers-contrast](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast) (detecting MacOS contrast preferences), as shown in the next example:
+
+Check `prefers-color-scheme` for `dark` or `light`,  `forced-colors` for `active` and `prefers-contrast` for `more`, and apply one of the available high contrast themes (Horizon High Contrast White or Horizon High Contrast Black)
+
+```ts
+    import { setTheme } from "@ui5/webcomponents-base/dist/config/Theme.js";
+
+	const darkMode = window.matchMedia("(prefers-color-scheme: dark)").matches;
+	const contrastMode = window.matchMedia("(forced-colors: active)").matches;
+	const prefersContrast = window.matchMedia("(prefers-contrast: more)").matches;
+
+	if (contrastMode || prefersContrast) {
```

Comment:
I'm wondering if we should align the media queries and the theme derivation with the ones defined in the theming base content:
https://github.com/SAP/theming-base-content/blob/master/content/Base/baseLib/sap_horizon_set/css_variables.css

The theming base content regards the `prefer-contrast` values `more` and `custom` as "the same".
The same is happening for the values `no-preference` and `less`.

Reply:
Hi @Thodd 

I wasn't able to force-match `prefer-contrast: custom`, but  from reading its description it seems like it should be treated same as "more", so I've added it to the examples.

At the same, in the latest iteration in the PR, I removed "forced-colors" usage, as my testing showed that "prefers-contrast: more" detects both MacOS increase contrast setting and Windows High Contrast Themes (Aquatic, Dessert..) and one should be fine without checking agains "forced-colors". Do you have different observations on that?

Reply:
Hi @ilhan007,

yes I came to the same conclusion.
`forced-colors` does not seem to add any benefit.
I also couldn't get the `custom` value for `prefers-contrast` 🤔
But then I only tested this on MacOS. I will ask another colleague to test this on their Windows machine too. 

I tried all kinds of variations now and this snippet is what gave me the best results for selecting a fitting theme. This also aligns with the theming base content. If I did this correctly, the theme selection should be mutual exclusive and we should always end up with a fixed theme:

```js
const allThemes = {
	"sap_horizon": window.matchMedia("only screen and (prefers-color-scheme: light) and (prefers-contrast: no-preference), screen and (prefers-color-scheme: light) and (prefers-contrast: less)").matches,
	"sap_horizon_dark": window.matchMedia("only screen and (prefers-color-scheme: dark) and (prefers-contrast: no-preference), screen and (prefers-color-scheme: dark) and (prefers-contrast: less)").matches,
	"sap_horizon_hcw": window.matchMedia("only screen and (prefers-color-scheme: light) and (prefers-contrast: custom), screen and (prefers-color-scheme: light) and (prefers-contrast: more)").matches,
	"sap_horizon_hcb": window.matchMedia("only screen and (prefers-color-scheme: dark) and (prefers-contrast: custom), screen and (prefers-color-scheme: dark) and (prefers-contrast: more)").matches
};

// only one of the themes should be marked as true
const theme = Object.keys(allThemes).find((name) => {
	return allThemes[name];
});

console.log(`theme: ${theme}`);
```

Reply:
Thanks @Thodd, in case you find anything in addition, feel free to get in touch and suggest changes. For now I think this is good to go and will be merged.